src/connectionimpl.cpp

```diff
@@ -123,7 +123,9 @@ uint64_t ConnectionImpl::parse(const Buffer &buffer)
         try
         {
             // try to recognize the frame
-            ReceivedFrame receivedFrame(ReducedBuffer(buffer, processed), _maxFrame);
```

Comment:
You are making copy of object here `B bbb = B( A() );`. It is completely different situation and does not belong to what you have in your patch.